tasks/deploy/update.js

```diff
@@ -32,8 +33,23 @@ module.exports = function (gruntOrShipit) {
       shipit.releasePath = path.join(shipit.releasesPath, shipit.releaseDirname);
 
       shipit.log('Create release path "%s"', shipit.releasePath);
-      return shipit.remote('mkdir -p ' + shipit.releasePath)
-      .then(function () {
+      function mkdir() {
+        return shipit.remote('mkdir -p ' + shipit.releasePath);
+      }
+
+      var returnedPromise;
+      if (shipit.config.copyReleaseBeforeUpdate) {
+        returnedPromise = shipit.remote(util.format('set -o pipefail && ' +
```

Comment:
We need to be sure that this line is correct, and please put it in a separated function to clarify code and to add a JSDoc comment on it to explain what it does.